File: apphub-frontend/src/modules/custom/elite_base/common/component/landing_pad_selector/LandingPad.js
import { SelectOption } from "../../../../../../common/component/input/SelectInput";

const LandingPad = {
    SMALL: "SMALL",
    MEDIUM: "MEDIUM",
    LARGE: "LARGE",

    getOptions: () => {
        return [
            new SelectOption("S", LandingPad.SMALL),
            new SelectOption("M", LandingPad.MEDIUM),
            new SelectOption("L", LandingPad.LARGE),
        ];
    }
}

export default LandingPad;

File: apphub-integration/src/main/java/com/github/saphyra/apphub/integration/structure/api/elite_base/LandingPad.java
package com.github.saphyra.apphub.integration.structure.api.elite_base;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum LandingPad {
    SMALL(1),
    MEDIUM(2),
    LARGE(3);

    private final int size;

    public boolean isLargeEnough(LandingPad minLandingPad) {
        return size >= minLandingPad.size;
    }
}
